six/modules/c++/samples/get_output_plane_image.cpp

```diff
@@ -0,0 +1,159 @@
+     * image, you will have to mitigate this. When viewing in MATLAB,
+     * for example, this may be done by
+     * imagesc(read_sio('outputPlane.sio', [0, 50]));
+     */
+    try
+    {
+        cli::ArgumentParser parser;
+        parser.setDescription("Read a SICD, project the image data to the "
+                "output plane, and write it as an SIO");
+        parser.addArgument("-x --polyOrderX", "Order for x-direction polynomials",
+                           cli::STORE, "polyOrderX", "POLY_ORDER_X", 1, 1)->
+                           setDefault(3);
+        parser.addArgument("-y --polyOrderY", "Order for y-direction polynomials",
+                           cli::STORE, "polyOrderY", "POLY_ORDER_Y", 1, 1)->
+                           setDefault(3);
+        parser.addArgument("input", "Input SICD", cli::STORE, "input", "INPUT",
+                            1, 1);
+        parser.addArgument("output", "Output SIO Pathname", cli::STORE,
+                           "output", "OUTPUT", 1, 1);
+
+        const std::auto_ptr<cli::Results> options(parser.parse(argc, argv));
+
+        const std::string sicdPathname(options->get<std::string>("input"));
+        const std::string outputPathname(options->get<std::string>("output"));
+        const size_t polyOrderX(options->get<size_t>("polyOrderX"));
+        const size_t polyOrderY(options->get<size_t>("polyOrderY"));
+        const std::vector<std::string> schemaPaths;
+
+        six::XMLControlRegistry registry;
+        registry.addCreator(six::DataType::COMPLEX,
+                new six::XMLControlCreatorT<six::sicd::ComplexXMLControl>());
+
+        std::auto_ptr<six::sicd::ComplexData> complexData;
+        std::vector<std::complex<float> > buffer;
+        six::sicd::Utilities::readSicd(sicdPathname, schemaPaths, complexData,
+                buffer);
+
+        // Derive AreaPlane if not defined
+        if (!six::sicd::AreaPlaneUtility::hasAreaPlane(*complexData))
+        {
+            six::sicd::AreaPlaneUtility::setAreaPlane(*complexData);
+        }
+        const six::sicd::AreaPlane& plane =
+                *complexData->radarCollection->area->plane;
+
+        // Grab outputToSlant plane polynomials
+        six::Poly2D toSlantRow;
+        six::Poly2D toSlantCol;
+        findOutputToSlantPolynomials(*complexData, polyOrderX, polyOrderY,
+                toSlantRow, toSlantCol);
+
+        mem::ScopedArray<float> outputArray(new float[
+                plane.xDirection->elements * plane.yDirection->elements]);
+        // Iterate over output plane, grabbing the appropriate input point
+        for (size_t yy = 0; yy < plane.yDirection->elements; ++yy)
+        {
+            const six::Poly1D rowPoly = toSlantRow.atY(yy);
+            const six::Poly1D colPoly = toSlantCol.atY(yy);
+            for (size_t xx = 0; xx < plane.xDirection->elements; ++xx)
+            {
+                const size_t line = math::round(rowPoly(xx));
```

Comment:
Hmm, so `rowPoly` and/or `colPoly` could return a negative number.  Granted that'll actually get handled here because you'll cast to a `size_t` which'll set it to a bazillion but I think it would be more explicit - particularly since the point is to help people understand how to use these polys` - if you check for that